### ISSUE TITLE
Bolus initiated: User unsure if user error or Loop bug

### ISSUE THREAD

**Issue by marydaisy** [LoopKit/Loop#1313]
I opened my Loop app and "caught" it delivering a 3 unit bolus I don't recall initiating. I don't know how to interpret the report to investigate if it was a user error or Loop. I have used Loop for 3.5 years and have never had this happen. I did confirm the bolus was being delivered on the pump and was able to cancel the bolus mid-delivery. It was clearly a bolus initiated by the Loop app vs the pump, as it was counting up on the Loop screen (vs the delay I see on Loop screen if I manually bolus on my Minimed pump).  I've been having general "buggy" issues with Loop lately, this being the most concerning. 
[Loop Issue Report.pdf](https://github.com/LoopKit/Loop/files/4507126/Loop.Issue.Report.pdf)

![image](https://user-images.githubusercontent.com/9502542/79814362-ea41b680-8332-11ea-8841-6973a0ec5018.png)

**Phone**
 - Hardware: iPhone 11
 - OS Version: iOS 13.3.1

**Loop Version**
 - Version Number: v2.1.0
 - Repo: I don't know what this means

**CGM**
 - Device: Dexcom G6

**Pump**
 - Manufacturer: Medtronic
 - Model: 722
 - Firmware version:

**Comment by Kdisimone** [comment]
Do you recall saving the 12g of carbs? In the Issue Report, 12g of carb entry was saved at 06:15:24 GMT and the bolus began at 06:15:42 GMT (so a little less than 20 second after the carb entry was saved). Do you remember entering the 12g and then trying to cancel out of the bolus delivery screen? Would seem like the bolus is associated with that carb entry?

**Comment by marydaisy** [comment]
Yes I definitely entered 12 carbs and then tried to cancel out of the bolus screen!

**Comment by marydaisy** [comment]
So it looks like I entered 12 carbs and delivered a 3 unit bolus for the 12 carbs? I mean that would be much more insulin than I would need for 12 carbs, but makes me feel better it was associated with me being in there. I also have face id now, so me just holding up the phone near my face could have confirmed it, which I'm not used to...

**Comment by Kdisimone** [comment]
Yeah, I think unfortunately, that food entry came in just on that little blip of an uptick on the CGM (right before crashing down)...so BG momentum and RC probably were helping predict a higher bolus need (vs carb ratio alone typically would've seen).  And I'm guessing *guessing* that the cancelation of bolus didn't go as planned and instead started delivering. So, probably some user interface improvements needed then to prevent people from accidentally bolusing instead of canceling. I do know the new carb entry tool has a button that boldly says "save without bolus" now to make it more clear that the carbs will save and no bolus delivered. The new screens are supposed to help make it more obvious that a bolus is or is not going to be delivered.

**Comment by Kdisimone** [comment]
Actually, i should look at the timestamps a little. Looks like the bolus timing is after those dropping CGM data points. I know there was talk about having the carb entry tool and bolus tool update if a new CGM data point comes in while the person is still deciding on a bolus/carb entry. That would've helped a lot in this situation perhaps. If BGs were dropping in the time between starting your carb entry and bolus delivery...that would've really been the difference here. Could've started the entry on the upswing of BGs and CGM dropped while still determining bolus recommendation.

**Comment by marydaisy** [comment]
So I noted the big drop and negative BG prediction and then ate and then entered the food to Loop, so yes I do know that BG drop was already noted in Loop when I bolused for the 12 carbs. The only thing I can think of us I mindlessly typed 3 units in? Because there's no way it would have recommended 3 units for 12 carbs. Even on my most insulin resistant days, my carb ratio is 1:9 at the very strongest for that time of night. And with that much IOB and a predicted BG below zero and only 12 carbs entered....

**Comment by marydaisy** [comment]
But yes, CGM had me jump way up almost immediately after this (way before the food even could have hit me). So the big drop was likely a compression drop and false. Maybe that's part of this whole issue is the rapid drop and rise?

**Comment by marydaisy** [comment]
Oh and as I'm building Loop reading your instructions, I realized I might have misunderstood your question about entering the 12 carbs and canceling out of the bolus screen. So just to be sure...here is what I did. I entered the 12 carbs intending for Loop to simply "save" them, knowing it would not recommend a bolus, but wanting it to account for them in the algorithm. I don't believe a bolus was recommended, nor was I anticipating one. So as far as I know I simply entered the 12 carbs, saved, and was done. Then I looked at the Loop screen a minute later and saw a bolus counting up. Confirmed the pump screen was counting up as well (so the pump was delivering said bolus), cancelled the bolus, and issued a report. :)
